geoportal/geoportailv3_geoportal/static-ngeo/js/apps/MainController.js

```diff
@@ -1374,15 +1375,38 @@ const MainController = function(
       { immediate: true }
     )
     const { bgLayer, layers } = storeToRefs(useMapStore())
+    const { config } = storeToRefs(useThemeStore())
+    const mapService = useMap()
+
     // monitor changes in layers to update state of mymaps
+    // Must wait theme definition to use useOpenLayers().getLayerFromCache()
+
     watch(
-      layers,
-      (layers) => {
-        this['selectedLayers'] = layers.map((l) => useOpenLayers().getLayerFromCache(l)).reverse();
-        this.compareLayers_();
-        $scope.$digest();
+      [layers, config],
+      ([layers, config], [oldLayers, oldConfig]) => {
+        if (config) { // Wait for themes to be loaded, and layer cache initialized
+          this['selectedLayers'] = layers.map((l) => useOpenLayers().getLayerFromCache(l)).reverse();
```

Comment:
maybe put a const useOpenlayers() before the map loop, so that the function does not have to be called at each iteration ?

Reply:
ok, this was not part of the change, sorry. I leave the remark, just in case

Reply:
good point!